### PR TITLE
46 create manage group screen

### PR DESCRIPTION
**Pull request by SteveLorentzen**

### CODE REVIEW

File: src/components/AddExisitingStudentsModal.tsx [hunk 1]
```diff
@@ -81,34 +83,60 @@ const AddExistingStudentsModal: React.FC<AddExistingStudentsModalProps> = ({ onC
...
+    let studentIsInStudentGroup = false
+    studentsInThisStudentGroupDocuments?.forEach(studentInGroup => {
+      if (studentInGroup.studentId === student.docId) {
+        studentIsInStudentGroup = true
+      }
     })
+    return !studentIsInStudentGroup
+  })
+
+  const selectAllHandler = (e: React.SyntheticEvent) => {
```
Comment (trevorfehrman) [inline src/components/AddExisitingStudentsModal.tsx:96]: We don't have to use synthetic events anymore per https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript

File: src/components/StudentPreview.tsx [hunk 2]
```diff
@@ -1,50 +1,78 @@
...
+  //   if (!isSelected) {
+  //     updatedStudentsToAdd = [...selectedStudentsToAdd, { studentId: docId, studentName, profilePic, studentFacts }]
+  //   } else {
+  //     updatedStudentsToAdd = selectedStudentsToAdd.filter(student => {
+  //       return student.studentId !== docId
+  //     })
+  //   }
+  //   setSelectedStudentsToAdd(updatedStudentsToAdd)
+  //   setIsSelected(!isSelected)
+  // }
```
Comment (trevorfehrman) [inline src/components/StudentPreview.tsx:35]: We keeping this around for some reason?

File: src/screens/ManageGroup.tsx [hunk 3]
```diff
@@ -1,19 +1,156 @@
...
+    const selection: string[] = []
+    if (selectedStudents.length < studentsInThisStudentGroupDocuments.length) {
+      studentsInThisStudentGroupDocuments.forEach(studentInThisGroup => {
+        selection.push(studentInThisGroup.docId)
+      })
+    }
+    setSelectedStudents(selection)
+  }
+
+  console.log(selectedStudents)
```
Comment (trevorfehrman) [inline src/screens/ManageGroup.tsx:44]: hey der lil guy, what r u doin here

**Review by trevorfehrman — APPROVED** [review]

File: src/screens/ManageGroup.tsx (hunk 3 above)
Comment (SteveLorentzen) [reply src/screens/ManageGroup.tsx:44]: Lol, I'll remove that console log

File: src/components/StudentPreview.tsx (hunk 2 above)
Comment (SteveLorentzen) [reply src/components/StudentPreview.tsx:35]: Nope.  Will delete.

File: src/components/AddExisitingStudentsModal.tsx (hunk 1 above)
Comment (SteveLorentzen) [reply src/components/AddExisitingStudentsModal.tsx:96]: That's a cool article, but I don't see a part that discusses no longer needing synthetic events.  What should it be typed as?

**Pull request merged**